packages/node-core/src/utils/autoQueue.spec.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {AutoQueue} from './autoQueue';
+import {delay} from './promise';
+
+describe('Auto Queue', () => {
+  it('create and flush AutoQueue', () => {
+    const newQ = new AutoQueue<void>(5);
+
+    const task1 = () => delay(5);
+    const task2 = () => delay(1);
+    const task3 = () => delay(10);
+
+    newQ.put(task1);
+    newQ.put(task2);
+    newQ.put(task3);
+
+    expect(newQ.size).toEqual(3);
+
+    newQ.flush();
+
+    expect(newQ.size).toEqual(0);
```

Comment:
Maybe we can check pendingPromise is false and that we can restart the autoqueue if a new task is pushed